prometheus-to-sd/config/source_config.go

```diff
@@ -112,7 +119,28 @@ func parseSourceConfig(uri flags.Uri, podId, namespaceId string) (*SourceConfig,
 	if err != nil {
 		return nil, err
 	}
-	return newSourceConfig(component, protocol, host, port, path, *auth, whitelisted, metricsPrefix, podConfig, whitelistedLabelsMap)
+	return newSourceConfig(component, protocol, host, port, path, *auth, whitelisted, metricsPrefix, podConfig, whitelistedLabelsMap, customResource, customLabels)
+}
+
+func getMap(v url.Values, name string) map[string]string {
+	n := len(name)
+	m := make(map[string]string)
+	for k, vals := range v {
+		if len(k) <= n {
+			continue
+		}
+		if k[:n] == name && k[n] == '[' && k[len(k)-1] == ']' {
```

Comment:
I prefer to use index to avoid making an error in regex :-)

But seriously, this is what I added tests for. If you think test coverage is insufficient here, let me know, I'll add more tests.